typing/mode.ml

```diff
@@ -2928,11 +2973,11 @@ let value_to_alloc_r2l m =
   { comonadic; monadic }
 
 module Modality = struct
-  type ('m, 'a) raw =
-    | Meet_with : 'a -> (('a, 'l * 'r) mode_comonadic, 'a) raw
-    | Join_with : 'a -> (('a, 'l * 'r) mode_monadic, 'a) raw
```

Comment:
This GADT was intended to encode the fact that `Meet_with` can be only applied on comonadic axes, to remove the assertion at L2988. But it doesn't seem to be work, so I'm simplify it.